### DISCUSSION TITLE
Lost & Found in the Command Line Homework

### DISCUSSION THREAD

**Discussion by ZoeLeBlanc** [CultureAsData-UIUC/is310-fall-2024#2] [Assignments]
Please add a link below to your Lost & Found in the Command Line Homework [https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/06-intro-file-formats#homework-lost--found-in-the-command-line](https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/06-intro-file-formats#homework-lost--found-in-the-command-line). Remember this assignment has two parts: creating your own maze and solving someone else's maze.

**Comment by tinayang922** [comment]
https://github.com/tinayang922/is310-coding-assignments/tree/main/command-line-maze

**Reply by ryanpearlman1** [reply]
<img width="286" alt="Screenshot 2024-09-08 at 9 48 08 PM" src="https://github.com/user-attachments/assets/6272aed1-3d5a-40c3-b569-1cde55808951">

**Reply by haydnelm** [reply]
![solution1](https://github.com/user-attachments/assets/cfa7d432-fb6c-40bf-9128-80092ff6171d)

**Reply by frangara** [reply]
![image](https://github.com/user-attachments/assets/0be2820e-5970-4c6b-b4ce-54a17f51a4d4)

**Reply by yunchaecho2** [reply]
<img width="259" alt="Screenshot 2024-09-11 at 10 47 16 PM" src="https://github.com/user-attachments/assets/ae5733f3-a668-44d6-a26e-757f145004ce">

**Comment by haydnelm** [comment]
https://github.com/haydnelm/is310-coding-assignments/tree/main/command-line-maze

**Reply by tinayang922** [reply]
<img width="576" alt="Screenshot 2024-09-08 at 10 59 58 PM" src="https://github.com/user-attachments/assets/b06f24d3-6fc3-4609-84ac-d5138d047fad">

**Comment by ryanpearlman1** [comment]
https://github.com/ryanpearlman1/is310-coding-assignments/blob/main/maze.zip

**Reply by tinayang922** [reply]
<img width="292" alt="Screenshot 2024-09-08 at 10 48 45 PM" src="https://github.com/user-attachments/assets/baf50ecd-15bd-4011-8ed7-ddc80f26dfb5">

**Reply by yosefstifanos** [reply]
first try: <img width="290" alt="image" src="https://github.com/user-attachments/assets/bd7c5ca9-f9c0-4488-ac75-75b6bcd6eca0">


second try: <img width="336" alt="image" src="https://github.com/user-attachments/assets/18407cf6-2be7-4aa2-9adc-3b6a3465b30c">

**Comment by averyWebster** [comment]
https://github.com/averyWebster/is310-coding-assignments/blob/main/command-line-maze.zip

**Reply by haydnelm** [reply]
![solution2](https://github.com/user-attachments/assets/b2d59ce7-9578-4c28-8a49-3bb865d3a606)

**Reply by OctaneCodes** [reply]
<img width="935" alt="Screenshot 2024-09-11 153516" src="https://github.com/user-attachments/assets/431df9b0-693f-45d7-a47c-28cf6831df6e">

**Reply by ZoeLeBlanc** [reply]
Overall, this looks great @averyWebster ! One small thing though is that your `.zip` file is not in the `command-line-maze` folder. If you could just move that and then do our normal git workflow (so `git add .` to add the changes, `git commit -m "moved maze" to commit the changes, and then `git push origin main` to make them live on GitHub) that would be perfect 👍

**Comment by yosefstifanos** [comment]


**Reply by kohtas2** [reply]
![screenshot](https://github.com/user-attachments/assets/30b76722-73e2-4f6a-940c-2b46de289255)

**Reply by esr27** [reply]
<img width="355" alt="Screenshot 2024-09-09 at 6 14 09 PM" src="https://github.com/user-attachments/assets/13819456-e4e5-4b61-beff-3b9b1d713be5">

**Reply by yosefstifanos** [reply]
> ![screenshot](https://private-user-images.githubusercontent.com/143128416/365799374-30b76722-73e2-4f6a-940c-2b46de289255.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5OTA5OTgsIm5iZiI6MTcyNTk5MDY5OCwicGF0aCI6Ii8xNDMxMjg0MTYvMzY1Nzk5Mzc0LTMwYjc2NzIyLTczZTItNGY2YS05NDBjLTJiNDZkZTI4OTI1NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMFQxNzUxMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNGIyY2U1M2FkNWNjOGQ3NjRjMjAxZTFiZmIyMzhjYWQwZDM0N2ZhYjdiZmZiMjBmNTBmYmFkOTkzNWYyYjJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.8htAeSN1W-pP1g_rrXX-9LYnRBo5iyZA93dzm9eMYjU)

redo it hehe. theres a secret file located somewhere

**Reply by yosefstifanos** [reply]
> <img alt="Screenshot 2024-09-09 at 6 14 09 PM" width="355" src="https://private-user-images.githubusercontent.com/143037245/365839821-13819456-e4e5-4b61-beff-3b9b1d713be5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5OTEyMTQsIm5iZiI6MTcyNTk5MDkxNCwicGF0aCI6Ii8xNDMwMzcyNDUvMzY1ODM5ODIxLTEzODE5NDU2LWU0ZTUtNGI2MS1iZWZmLTNiOWIxZDcxM2JlNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMFQxNzU1MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YmE1NjNhYWMyMjE4ODdjNWVhOTM0N2QxNWIwZTdhNTQxOWZkZWEzYTY2NjFkZDA4OTg5ZThhZTc0MjZlZGUwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Q5SImjnYy7R2bUb8YZ3Fr8d8dbYNqiOQ0_HU4nmw3zw">

redo it hehe. theres a secret file located somewhere

**Reply by Cindy1226** [reply]
<img width="321" alt="Screen Shot 2024-09-11 at 12 27 38 AM" src="https://github.com/user-attachments/assets/936fb9e6-7103-4aaa-b031-424ff63796f4">

**Comment by kohtas2** [comment]
https://github.com/kohtas2/is310-coding-assignments/blob/main/command-line-maze.zip

**Reply by ZoeLeBlanc** [reply]
![Screenshot 2024-09-21 at 5 41 03 PM](https://github.com/user-attachments/assets/17484283-7101-448d-be10-260e60600a9c)

Great work @kohtas2 👍 !

**Comment by esr27** [comment]
[`https://github.com/esr27/is310-coding-assignments/tree/main/command-line-maze`](https://github.com/esr27/is310-coding-assignments/blob/main/command-line-maze.zip)

**Reply by adg77** [reply]
<img width="356" alt="Screenshot 2024-09-11 at 11 28 02 PM" src="https://github.com/user-attachments/assets/70f18db1-bca4-47aa-81de-cdf3113f0fac">

**Reply by ZoeLeBlanc** [reply]
Hi @esr27 ! Overall this looks great. One small issue is that it looks like you zipped your `command-line-maze` folder which means your zip file is not in the folder. I would recommend renaming your current `command-line-maze` folder something like `door-maze` and then zipping that so you have `door-maze.zip` and then create a new `command-line-maze` folder and move that zip file into that folder. Happy to help if you have questions but very impressed with the sheer number of doors you created 👍

**Comment by Cindy1226** [comment]
https://github.com/Cindy1226/is310-coding-assignments/blob/main/command-line-maze/command-line-maze.zip

**Reply by averyWebster** [reply]
<img width="315" alt="Screenshot 2024-09-10 at 9 11 00 AM" src="https://github.com/user-attachments/assets/7d08fef1-46d0-4dde-8583-1b98562d982d">

**Reply by GloriaLiu930** [reply]
![exit](https://github.com/user-attachments/assets/6550b5f9-4bef-4ea5-b21a-c24aa68ff25c)

**Reply by theajk2** [reply]
[
<img width="340" alt="cs" src="https://github.com/user-attachments/assets/aec98c38-c4f1-4a34-bda6-1c20daf4c9ba">
](url)

**Reply by jasonmorris8** [reply]
<img width="1208" alt="Screenshot 2024-09-11 at 4 44 41 PM" src="https://github.com/user-attachments/assets/cf084e3e-31d3-4730-bdfc-200a6361a202">

**Comment by GloriaLiu930** [comment]
https://github.com/GloriaLiu930/is310-coding-assignments/tree/main/command-line-maze

**Reply by officalcharles** [reply]
<img width="880" alt="Screenshot 2024-09-10 at 11 24 30 PM" src="https://github.com/user-attachments/assets/713736e0-ae45-4e5a-bf2f-429daf877592">

**Comment by officalcharles** [comment]
https://github.com/officalcharles/is310-coding-assignments/tree/master/maze

**Comment by yosefstifanos** [comment]
https://github.com/yosefstifanos/is310-coding-assignments/blob/main/command-line-maze.zip

**Reply by yosefstifanos** [reply]
*Redo of my txt files to add further information*

**Reply by yosefstifanos** [reply]
@ZoeLeBlanc https://github.com/CultureAsData-UIUC/is310-fall-2024/discussions/2#discussioncomment-10594159

**Reply by yosefstifanos** [reply]
I dont know if they want to redo it or not, but I didnt add in the comment to find the actual hidden text.

**Comment by AlanK3** [comment]
https://github.com/AlanK3/is310-coding-assignments/tree/main/command-line-maze
Should be straightforward

**Reply by ahmadsawwan** [reply]
Bro did not just say straightforward..
<img width="600" alt="Screen Shot 2024-09-11 at 2 15 43 PM" src="https://github.com/user-attachments/assets/c1c03de5-0d49-4609-9485-20687e45a90f">


PS. I had to cheat, I was lost for a long time..

**Comment by ahmadsawwan** [comment]
https://github.com/ahmadsawwan/ahmadsawwan-is310-coding-assignments/tree/main/command-line-maze

read instructions (:

**Reply by lhayedimarucut** [reply]
<img width="553" alt="Screenshot 2024-09-11 at 3 57 38 PM" src="https://github.com/user-attachments/assets/aabfc497-91a6-464b-b324-7a6e1cd47636">

**Reply by SamuelSokolovsky** [reply]
<img width="486" alt="Screenshot 2024-09-11 at 5 17 47 PM" src="https://github.com/user-attachments/assets/10bf5e60-47be-487b-9ec7-d6abffec488d">

**Comment by frangara** [comment]
https://github.com/frangara/is310-coding-assignments/tree/main/command-line-maze 
Francis Angara

I just made this! This is themed after one of my favorite shows Naruto so be sure to brush up on this show if you plan on solving this!

**Reply by chuzesta** [reply]
<img width="576" alt="bruh" src="https://github.com/user-attachments/assets/52709f1e-d059-49ed-8060-8a9ee2c4486f">

**Comment by theajk2** [comment]
https://github.com/theajk2/is310-coding-assignments/tree/main/command%20line%20maze

**Reply by ZoeLeBlanc** [reply]
![Screenshot 2024-09-21 at 6 06 26 PM](https://github.com/user-attachments/assets/d72bd7ac-0e2f-4f8e-8483-8fa5b71e524a)

Great work @theajk2 👍 !

**Comment by lhayedimarucut** [comment]
https://github.com/lhayedimarucut/is310-coding-assignments/tree/main/command-line-maze

**Reply by divyaj5** [reply]
<img width="498" alt="image" src="https://github.com/user-attachments/assets/0d986e48-b4ef-4e47-859a-60199c69b0ff">

**Comment by nchat3** [comment]
https://github.com/nchat3/is310-coding-assignments.git

**Comment by SamuelSokolovsky** [comment]
https://github.com/SamuelSokolovsky/is310-coding-assignments/tree/main/command-line-maze

**Reply by ZoeLeBlanc** [reply]
![Screenshot 2024-09-21 at 5 58 49 PM](https://github.com/user-attachments/assets/97a218b6-2f0b-4cba-86ab-5efaad4b7a25)

Great work @SamuelSokolovsky 👍

**Comment by OctaneCodes** [comment]
https://github.com/OctaneCodes/is310-coding-assignments/tree/main/command-line-maze

**Reply by Rebe-Rod** [reply]
![Screenshot 2024-09-11 221301](https://github.com/user-attachments/assets/2557531f-0958-41a5-9383-58f2f9b20b04)

**Comment by yunchaecho2** [comment]
https://github.com/yunchaecho2/is310-coding-assignments

**Reply by ZoeLeBlanc** [reply]
Hi @yunchaecho2 ! Good work on your maze but noticed that your solution maybe doesn't quite work.

Here's what I see when I end the maze:

![Screenshot 2024-09-21 at 6 14 01 PM](https://github.com/user-attachments/assets/265ee726-0191-4551-a7c0-6fd2b4d7974c)

And if I try to open the file, I see this notice:

![Screenshot 2024-09-21 at 6 13 43 PM](https://github.com/user-attachments/assets/89b979a3-f4f0-4a74-93ec-38323559b453)

It seems like maybe you tried to create a password protected file? I did find your `.hidden_file.txt` so got the password but didn't get a chance to use it. Hope this feedback helps and happy to answer any questions you have 👍

**Reply by yunchaecho2** [reply]
Professor Le Blanc, thank you for your feedback! Based on the given feedback, I've created new maze for this assignment. https://github.com/yunchaecho2/is310-coding-assignments/tree/main/command-line-maze2

**Comment by Rebe-Rod** [comment]
https://github.com/Rebe-Rod/is310-coding-assingments

**Reply by ZoeLeBlanc** [reply]
Hi @Rebe-Rod ! Great work on this maze. One issue is that I couldn't find the center of the maze 😅 . That might just be on me but wanted to flag it for you just in case.

I also noticed two small issues in your repo. First, you might consider moving your `command-line-maze.zip` file into your `command-line-maze` folder. Second, I noticed that you have an is310-coding-assignments folder with a small white arrow in your repo, like the image below:

<img width="1004" alt="Screenshot 2024-09-21 at 5 52 26 PM" src="https://github.com/user-attachments/assets/de2d3f98-513b-4d96-a80c-5b3facb60ac5">

That's a sign that you had accidentally created a `.git` folder in that folder with `git init`, and so it is now showing as a `git submodule.` You can fix this locally and I would be happy to meet with you to help you complete these steps or chat over Discord, so don't hesitate to reach out if that would be helpful.

Here are the steps to remove `git submodules`:

1. **Deinitialize the Submodule:**
   - Open your terminal or command line in the root directory of the repository (so in `is310-coding-assignments`.
   - Run this command to deinitialize the submodule (this breaks the submodule connection without deleting your folder):
   - The exact command will be:
     ```bash
     git submodule deinit -f is310-coding-assignments
     ```

2. **Remove the Submodule from Git's Index:**
   - Run the following command to remove the submodule from the Git index, but keep the folder and its content intact:
     ```bash
     git rm --cached is310-coding-assignments
     ```

3. **Remove the Submodule Entry from `.gitmodules`:**
   - Edit the `.gitmodules` file (if it exists) and manually remove the section that refers to the submodule.
   - Look for a section that looks like this:
     ```bash
     [submodule "path/to/submodule"]
       path = path/to/submodule
       url = URL_OF_SUBMODULE
     ```
   - Remove that section, save, and close the file.

4. **Commit the Changes:**
   - After these changes, commit them to your local repository:
     ```bash
     git add .
     git commit -m "Removed submodule but kept folder"
     ```

5. **Push the Changes:**
   - Push your changes to GitHub:
     ```bash
     git push origin main
     ``` 
   Overall good work though 👍

**Comment by adg77** [comment]
https://github.com/adg77/is310-coding-assignments/blob/main/command-line-maze.zip

**Reply by AG-Samaniego** [reply]
![image](https://github.com/user-attachments/assets/8ecb56c3-1637-4065-8cf8-952408cee842)
I know very little about football, but this was a really fun maze ngl I learned a lot!

**Reply by ZoeLeBlanc** [reply]
Hi @adg77 ! Overall the maze looks great but I noticed a few things for you to fix.

First, your maze should be in a `command-line-maze` folder in your `is310-coding-assignments`.

Second, I noticed your commit username is not your GitHub username. This is usually a sign that your local git is not correctly configured. If you follow the steps here [https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/01-course-tools/#git-configuration](https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/01-course-tools/#git-configuration) of adding in your exact GitHub username (so in your case it would be `adg77`) and then the email you use to sign into GitHub. 

Once you do that your commits should show up correctly. Happy to help if you have questions 👍

**Comment by JBrown-0** [comment]
https://github.com/JBrown-0/is310-coding-assignments/blob/main/README.md

had trouble pushing to github even with SSH key

**Reply by ZoeLeBlanc** [reply]
Hi @JBrown-0 ! This link gives me a `404` error so wondering if you accidentally deleted the file? Feel free to ping me when you add it back 👍

**Comment by henRotunno** [comment]
https://github.com/henRotunno/is310-coding-assignments/tree/main/command-line-maze/intro-maze

**Reply by ZoeLeBlanc** [reply]
![Screenshot 2024-09-21 at 5 22 16 PM](https://github.com/user-attachments/assets/75327978-c672-45cf-9a52-328e1e5a8996)

Great work @henRotunno 👍

**Comment by jasonmorris8** [comment]
https://github.com/jasonmorris8/is310-coding-assignments/tree/main/command-line-maze

**Reply by ZoeLeBlanc** [reply]
![Screenshot 2024-09-21 at 5 26 30 PM](https://github.com/user-attachments/assets/f5bc1c21-66b0-4b12-95b3-293defa41f33)

Great work @jasonmorris8 👍

**Comment by Zifanzhao0801** [comment]
https://github.com/Zifanzhao0801/is310-coding-assignments/tree/main/command-line-maze

**Reply by ZoeLeBlanc** [reply]
Hi @Zifanzhao0801 ! This is a good start but it looks like none of your `.txt` files have any text in them, except for the `clue.txt`. Maybe you forgot to add it in? Just flagging this for you to resolve and happy to answer questions if you're having any issues 👍 

![Screenshot 2024-09-21 at 6 25 29 PM](https://github.com/user-attachments/assets/5aa348f3-6b6e-45fe-82e3-81aa58f052ba)

**Comment by shoirathi** [comment]
https://github.com/shoirathi/is310-coding-assignments/blob/main/entrance.zip

**Reply by LucasWashor** [reply]
<img width="770" alt="CommandMaze" src="https://github.com/user-attachments/assets/0bbc4db5-6c90-4906-9e02-1ffbff5d0b0d">

**Reply by ZoeLeBlanc** [reply]
Hi @shoirathi ! Just flagging that I can't see your `command-line-maze` folder because of the issue I outlined in this comment [https://github.com/CultureAsData-UIUC/is310-fall-2024/discussions/1#discussioncomment-10698139](https://github.com/CultureAsData-UIUC/is310-fall-2024/discussions/1#discussioncomment-10698139) . Please don't hesitate to reach out for help if you face difficulties resolving the issue and please ping me here once you fix it 👍

**Comment by divyaj5** [comment]
https://github.com/divyaj5/is310-coding-assignments/tree/master/command-line-maze

**Reply by elee33** [reply]
<img width="518" alt="Untitled 2" src="https://github.com/user-attachments/assets/1b9c10bd-61d3-4ae4-aa5a-2da3fb16fc14">

**Comment by elee33** [comment]
https://github.com/elee33/is310-coding-assignments/blob/main/command-line-maze.zip

**Reply by ZoeLeBlanc** [reply]
Hi @elee33 ! When I try to open your command line maze it asks for a password? Also your maze should be in a folder called `command-line-maze` in your `is310-coding-assignments` folder. Happy to help if you have questions 👍

**Comment by LucasWashor** [comment]
https://github.com/LucasWashor/is310-coding-assignments/blob/main/maze.zip

**Reply by shoirathi** [reply]
![Screenshot 2024-09-17 200714](https://github.com/user-attachments/assets/68815bd4-99fe-4c8d-b370-5070c9aed990)

**Comment by AG-Samaniego** [comment]
Sorry this is late (i got really invested lol) (in retrospect I think I should've just made the story for Twine rather than CL but it was still fun!)
https://github.com/AG-Samaniego/is310-coding-assignments/tree/main/command-line-maze

**Reply by ZoeLeBlanc** [reply]
Very impressive CLI maze!! I will be honest I knew almost none of the songs but appreciated the playlists at the end. Again very well done 👍 

![Screenshot 2024-09-19 at 5 10 34 PM](https://github.com/user-attachments/assets/85ee34fb-82e9-43ed-911b-6e6d8dcd93b4)

**Reply by divyaj5** [reply]
<img width="1076" alt="image" src="https://github.com/user-attachments/assets/856caa9a-48e3-49cf-9439-8074ba8c53d3">

**Comment by chuzesta** [comment]
https://github.com/chuzesta/is310-coding-assignments

**Reply by ZoeLeBlanc** [reply]
![Screenshot 2024-09-21 at 5 31 54 PM](https://github.com/user-attachments/assets/5bde77e5-ba93-49e6-b5a2-90ee2aa36639)

Great maze @chuzesta ! One small thing is you'll notice that when I did `cat keys.rtf` there's a lot of weird characters that appear in that image. That's because you used `.rtf` which is a rich text format. I would recommend using either `.txt` or `.md` in the future to avoid that formatting. Also just as an FYI your directories had spaces which makes them tricky to navigate. Usually we put dashes or underscores in directories, since otherwise it is difficult for computers to know if they are multiple directories or the same one. Overall though good work 👍

**Comment by yukthas2** [comment]
https://github.com/yukthas2/is310-coding-assignments/tree/main/command-line-maze
